nanodbc/nanodbc.cpp

```diff
@@ -2028,6 +2041,27 @@ class statement::statement_impl
             NANODBC_THROW_DATABASE_ERROR(stmt_, SQL_HANDLE_STMT);
     }
 
+    void set_param_descr(
+        const std::vector<short>& idx,
+        const std::vector<short>& type,
+        const std::vector<unsigned long>& size,
+        const std::vector<short>& scale)
+    {
+
+        NANODBC_ASSERT(idx.size() == type.size());
+        NANODBC_ASSERT(idx.size() == size.size());
+        NANODBC_ASSERT(idx.size() == scale.size());
```

Comment:
Are you convinced validating user-specified input with assertions is the right approach?

In my opinion, an exception should be thrown.

Reply:
This is consistent with the strategy used elsewhere in the code, for example: https://github.com/nanodbc/nanodbc/blob/master/nanodbc/nanodbc.cpp#L1862

But yeah, maybe it could be revisited in the future in another PR.

Reply:
@lexicalunit Good catch. I think we should lean towards replacing (or complementing) such assertions with exceptions.

Recently, I have received some learning from Boost folks on (mis)using assertions for validating user-specified input :-) Here are some pointers on #boost channel of C++ Slack 

- https://cpplang.slack.com/archives/C27KZLB0X/p1564765315241300
- https://cpplang.slack.com/archives/C27KZLB0X/p1564783195259200?thread_ts=1564777009.256000&cid=C27KZLB0X